test/html_visitor_test.rb

```diff
@@ -1,52 +1,183 @@
 # frozen_string_literal: true
 require "test_helper"
 
-class HtmlVisitorTest < Minitest::Test
-  def setup
-    @tracer = TracerCheck.new
-    @visitor = ThemeCheck::HtmlVisitor.new(ThemeCheck::Checks.new([@tracer]))
-  end
-
-  def test_elements_with_variables
-    template = parse_liquid(<<~END)
-      <a href="/about">About</a>
-      <img src="a.jpg" width="{{ image.width }}" height="{{ image.height }}">
-    END
-    @visitor.visit_template(template)
-    assert_equal([
-      :on_document,
-      :on_element,
-      :on_a,
-      :on_text, "About",
-      :after_a,
-      :after_element,
-      :on_text, "\n",
-      :on_element,
-      :on_img,
-      :after_img,
-      :after_element,
-      :on_text, "\n",
-      :after_document
-    ], @tracer.calls)
-  end
+module ThemeCheck
+  class HtmlVisitorTest < Minitest::Test
+    def setup
+      @tracer = TracerCheck.new
+      @visitor = HtmlVisitor.new(Checks.new([@tracer]))
+    end
 
-  def test_elements_with_liquid_tags
-    template = parse_liquid(<<~END)
-      {% capture x %}
+    def test_elements_with_variables
+      template = parse_liquid(<<~END)
         <a href="/about">About</a>
-      {% endcapture %}
-    END
-    @visitor.visit_template(template)
-    assert_equal([
-      :on_document,
-      :on_text, "{% capture x %}\n  ",
-      :on_element,
-      :on_a,
-      :on_text, "About",
-      :after_a,
-      :after_element,
-      :on_text, "\n{% endcapture %}\n",
-      :after_document
-    ], @tracer.calls)
+        <img src="a.jpg" width="{{ image.width }}" height="{{ image.height }}">
+      END
+      @visitor.visit_template(template)
+      assert_equal([
+        :on_document,
+        :on_element,
+        :on_a,
+        :on_text, "About",
+        :after_a,
+        :after_element,
+        :on_text, "\n",
+        :on_element,
+        :on_img,
+        :after_img,
+        :after_element,
+        :on_text, "\n",
+        :after_document
+      ], @tracer.calls)
+    end
+
+    def test_elements_with_liquid_tags
+      template = parse_liquid(<<~END)
+        {% capture x %}
+          <a href="/about">About</a>
+        {% endcapture %}
+      END
+      @visitor.visit_template(template)
+      assert_equal([
+        :on_document,
+        :on_text, "{% capture x %}\n  ",
+        :on_element,
+        :on_a,
+        :on_text, "About",
+        :after_a,
+        :after_element,
+        :on_text, "\n{% endcapture %}\n",
+        :after_document
+      ], @tracer.calls)
+    end
+
+    def test_elements_with_quotes_inside_quotes
+      @attribute_checker = HTMLAttributeIntegrityMockCheck.new
+      @visitor = HtmlVisitor.new(Checks.new([@attribute_checker]))
+      template = parse_liquid(<<~END)
+        <script href="{{ \"a.js\" | asset_url }}" defer></script>
```

Comment:
can we add a test with an example that doesn't escape its quotes? e.g.

`<script href="{{ "c.js" | asset_url }}" defer></script>`